doc/apiv3-documentation.apib

```diff
@@ -2271,229 +2274,792 @@ Lists the columns available for work packages of this project in the form of fie
-| regularExpression | The value of the property must match the given regular expression (if not null)    | String   | null    |
-| required          | If true this property is not nullable                                              | Boolean  | true    |
-| writable          | If false it is not allowed to **change** the properties value                      | Boolean  | true    |
++ Response 204 (application/hal+json)
 
-All of the above properties that do not have a default value *must* be present in the schema.
-For properties that have a default value, the client can assume the default value, if the property is missing.
+    Returned if the relation was deleted successfully.
+    The response body is empty.
 
-Note that regular expressions used in the API follow the rules of [Ruby Regular Expressions](http://www.ruby-doc.org/core-2.1.5/Regexp.html).
+    + Body
 
-## Example Schema [/api/v3/example/schema]
++ Response 403 (application/hal+json)
+
+    Returned if the client does not have sufficient permissions.
+
+    **Required permission:** manage work package relations
+
+    + Body
+
+            {
+                "_type": "Error",
+                "errorIdentifier": "urn:openproject-org:api:v3:errors:MissingPermission",
+                "message": "You are not allowed to delete this relation."
+            }
+
++ Response 404 (application/hal+json)
+
+    Returned if the relation does not exist or the client does not have sufficient permissions to see it.
+
+    **Required permission:** manage work package relations
+
+    + Body
+
+            {
+                "_type": "Error",
+                "errorIdentifier": "urn:openproject-org:api:v3:errors:NotFound",
+                "message": "The specified relation does not exist."
+            }
+
+## Relation schema [/api/v3/relations/schema]
 
 + Model
     + Body
 
             {
                 "_type": "Schema",
                 "_links": {
-                    "self": { "href": "/api/v3/example/schema" }
+                    "self": { "href": "/api/v3/relations/schema" }
                 },
 
-                "lockVersion": {
-                    "name": "Resource Version",
+                "id": {
+                    "name": "Relation ID",
                     "type": "Integer",
                     "writable": false
                 },
-                "subject": {
-                    "name": "Subject",
+                "type": {
+                    "name": "Relation Type",
                     "type": "String",
-                    "minLength": 1,
-                    "maxLength": 255
+                    "writable": true,
+                    "allowedValues": [
+                      "relates", "duplicates", "duplicated", "blocks", "blocked",
+                      "precedes", "follows", "includes", "partof", "requires", "required"
+                    ]
```

Comment:
For the work packages, we did not put the `allowedValues` into the not embedded schema. That way, the general schema can be referenced by multiple resources in their link section. As the `allowedValues` might depend on the status the referring object is in reusing the schema would otherwise be inadequate. For relations the same is true for the `delay` attribute which does not exist at all for some types. It is also true for the `to`and `from` attribute which's `allowedValues` will change depending on the work package already selected for one of the attributes. For user resources the same will be true again as it e.g. is not possible to set the status to `invited` again after the user is `active`

*From the trenches*
The same is of course true for other properties such as `writable`. E.g. it is not allowed to write to the `estimatedTime` property of a work package if said work package has children. The generic schema will still declare the field to be writable because we don't know better and the field might be writable. This allows us to differentiate in the work packages list between fields that will never be writable and those that might be writable. I'd welcome a better way of doing this and by that be able to get rid of all attributes except for `name` and `type` in the generic schema. Or we change the generic schema to always contain all values that might be available but that do not necessarily have to be for a specific resource. This might actually be preferable but we would have to change it for work packages. 

*TL;DR*
The generic schema only should not contain `allowedValues` as of now.